stdlib/public/stubs/LibcShims.cpp

```diff
@@ -98,33 +99,57 @@ int swift::_swift_stdlib_close(int fd) {
 #endif
 }
 
+#if !defined(_WIN32)
 // Guard compilation on the typedef for __swift_pthread_key_t in LibcShims.h
 // being identical to the platform's pthread_key_t
 static_assert(std::is_same<__swift_pthread_key_t, pthread_key_t>::value,
               "This platform's pthread_key_t differs. If you hit this assert, "
               "fix __swift_pthread_key_t's typedef in LibcShims.h by adding an "
               "#if guard and definition for your platform");
+#endif
 
 SWIFT_RUNTIME_STDLIB_INTERFACE
-int swift::_swift_stdlib_pthread_key_create(
+int swift::_swift_stdlib_tls_key_create(
   __swift_pthread_key_t * _Nonnull key,
   void (* _Nullable destructor)(void *)
 ) {
-  return pthread_key_create(key, destructor);
-}
-
-SWIFT_RUNTIME_STDLIB_INTERFACE
-void * _Nullable swift::_swift_stdlib_pthread_getspecific(
+  #if defined(_WIN32)
+    // We're using the Fls- (Fiber) instead of Tls- (Thread) functions since
+    // on Windows, "If no fiber switching occurs, FLS acts exactly the same 
+    // as thread local storage," and FlsAlloc takes a destructor while
+    // TlsAlloc doesn't.
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/ms682661.aspx
```

Comment:
This also makes the code more portable, as some Windows targets (e.g. Windows Store, Windows Phone) only support the FLS variant.